frameworks/OCaml/webmachine/src/src/bin/tfb.ml

```diff
@@ -160,35 +200,36 @@ let main () =
      | Some result -> result)
     >>= fun (status, headers, body, _) ->
     let headers = Header.add headers "Server" "webmachine" in
-    let headers = Header.add headers "Date" (Lib.Time.now ()) in
+    let headers = Header.add headers "Date" Lib.Time.(!memo_date) in
     Server.respond ~headers ~body ~status ()
   in
 
-  let config = Server.make ~callback () in
-  Server.create ~mode:(`TCP (`Port port)) config >|= fun () ->
-  Printf.eprintf "hello_lwt: listening on 0.0.0.0:%d%!" port
-
-let () =
-  let options =
-    [
-      ("fd_passing", `fd_passing);
-      ("fdatasync", `fdatasync);
-      ("get_affinity", `get_affinity);
-      ("get_cpu", `get_cpu);
-      ("get_credentials", `get_credentials);
-      ("libev", `libev);
-      ("madvise", `madvise);
-      ("mincore", `mincore);
-      ("recv_msg", `recv_msg);
-      ("send_msg", `send_msg);
-      ("set_affinity", `set_affinity);
-      ("wait4", `wait4);
-    ]
+  let ipaddr = Unix.inet_addr_any in
+  let sockaddr = Unix.ADDR_INET (ipaddr, port) in
+  let socket =
+    Lwt_unix.socket (Unix.domain_of_sockaddr sockaddr) Unix.SOCK_STREAM 0
   in
-  List.iter
-    (fun (str, opt) ->
-      print_endline ("option " ^ str ^ ": " ^ string_of_bool (Lwt_sys.have opt)))
-    options;
-  (* https://github.com/mirage/ocaml-cohttp/issues/328#issuecomment-222583580 *)
-  Lwt_io.set_default_buffer_size 0x10000;
-  Lwt_main.run (main ())
+  Lwt_unix.setsockopt socket Unix.SO_REUSEADDR true;
+
+  Lwt_main.run
+  @@ ( Lwt_unix.bind socket sockaddr >|= fun () ->
+       Lwt_unix.listen socket (Lwt_unix.somaxconn () [@ocaml.warning "-3"]) );
+
+  for i = 1 to nproc do
+    ignore @@ Lwt_io.flush_all ();
+    if Lwt_unix.fork () = 0 then (
+      (* child *)
+      Lib.Time.refresh_date ();
+      ignore
+      @@ Lwt_io.eprintf "Listening on %s:%d (child %d)\n"
+           (Unix.string_of_inet_addr ipaddr)
+           port i;
+      let config = Server.make ~callback () in
+      ignore @@ Server.create ~mode:(`TCP (`Socket socket)) config;
+      let forever, _ = Lwt.wait () in
+      Lwt_main.run forever;
+      exit 0 )
+  done;
+  Unix.pause ()
```

Comment:
You need to loop here, because any signal is gonna wake up the process

Reply:
It seems to work without a loop, what could happen?

With the loop all signals are effectively ignored, right?

Reply:
Well, if you want to keep the (parent) process running (for example if it's your monitored Docker process), you should loop to avoid exiting.

If you don't care about the parent process (as currently, since haproxy is actually the monitored Docker process), you can just remove `Unix.pause` altogether. You'd need to update it the code if you get rid of haproxy.

My main point was that the current version (`Unix.pause` without loop) looks unintended.